tools/nl/embeddings/utils.py

```diff
@@ -124,84 +75,123 @@ def get_model(catalog: Catalog, env: Env, model_name: str) -> EmbeddingsModel:
   return model
 
 
-def build_and_save_preindex(fm: FileManager) -> Tuple[List[str], List[str]]:
+def get_saved_embeddings(embeddings_path: str) -> List[SentenceObject]:
+  """Load saved embeddings from a CSV file."""
+  try:
+    if gcs.is_gcs_path(embeddings_path):
+      embeddings_path = gcs.maybe_download(embeddings_path)
+    df = pd.read_csv(embeddings_path)
+    sentences = []
+    for _, row in df.iterrows():
+      dcid = row['dcid']
+      sentence = row['sentence']
+      vector = row.drop(labels=['dcid', 'sentence']).astype(float).tolist()
+      sentences.append(SentenceObject(text=sentence, dcid=dcid, vector=vector))
+    return sentences
+  except Exception as e:
+    logging.error(e)
+    return []
```

Comment:
Do we want to continue on Exception (when does it happen?) vs. just throw up to user

Reply:
Want to continue with [] list, since this is helper list, so should not block the main flow. Worst case is to provide no help..(with empty list)